### ISSUE TITLE
ENH: How about "cannot reindex from an index with duplicate values"?

### ISSUE THREAD

**Issue by jabowery** [pandas-dev/pandas#40622]
#### Is your feature request related to a problem?

I spent quite a bit of time trying to figure out what this message means:

`*** ValueError: cannot reindex from a duplicate axis`

#### Describe the solution you'd like

It would have been much easier for me to figure out what I was doing wrong with the message:

`*** ValueError: cannot reindex from an index with duplicate values`

**Comment by rhshadrach** [comment]
Thanks @jabowery - I find your proposed message to be more clear than the existing. It is still a bit unclear to me which index the message is referring to. I might suggest something like:

    cannot use reindex on an index with duplicate values.

cc @jorisvandenbossche - any thoughts here?

**Comment by Tylermarques** [comment]
To add to this discussion, the error message would ideally be different when column names are identical, but your index is unique. I just spent a while chasing this as the message insinuates that my index is not unique even though the issue was duplicate column names.

**Comment by rhshadrach** [comment]
Agreed it would be helpful to mention index vs columns. I'm not sure how much awareness there is, but the columns of a DataFrame _are_ an Index:

```
print(pd.DataFrame(columns=['a', 'b']).columns)
Index(['a', 'b'], dtype='object')
```

@Tylermarques - would you be interested in working on a PR to improve this error message?

**Comment by Tylermarques** [comment]
@rhshadrach I don't have a ton of experience contributing, but I'll read the contributing guide and take a crack at it.

**Comment by Tylermarques** [comment]
take

**Comment by jbrockmendel** [comment]
I see "cannot reindex on an axis with duplicate labels".  which seems fine.
